src/spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -134,7 +165,7 @@ def _compute_metrics(self, sorting_analyzer, unit_ids=None, verbose=False, **job
 
         import pandas as pd
 
-        metrics = pd.DataFrame(index=unit_ids)
+        metrics = pd.DataFrame(index=sorting_analyzer.unit_ids)
```

Comment:
The unit_ds is super super important, because it allow to compute only a subsets in case of the merging process.
```suggestion
        metrics = pd.DataFrame(index=unit_ids)
```

Reply:
I agree this should be controlled by `unit_ids`, otherwise [this merging snippet](https://github.com/SpikeInterface/spikeinterface/blob/406f99bf7d2f68b68db3e4535ba6e004331de419/src/spikeinterface/qualitymetrics/quality_metric_calculator.py#L131-L133) will not behave as it should